### PR TITLE
Cache Bazel query into workspace tree

### LINKED ISSUE #94: Cache Analysis Bazel Workspace File Systems
The Bazel invoker created from #81 will help us understand how our client's Bazel projects are laid out. This is great, but the Bazel invoker takes a long time to complete. We need to cache our client's file system so we don't hang for so long.

Use the Bazel invoker to construct a WorkspaceTree, which is created from in #91. Cache the workspace tree after each Bazel query to save on performance.

- [x] Use the bazel invoker to interpret a client's bazel file system.
- [x] Initialize the analyzer on extension start. This should cache a representation of our user's file system as a `WorkspaceTree`
- [ ] Whenever appropriate, or possibly on `didUpdate` from the `BazelServices`, update the cached workspace stored in the analyzer. Only update what needs to be updated.

Unit tests
- [x] Create unit tests to test this behavior

### PR DESCRIPTION
**Pull request by NovaMachina**
This constructs the the workspace tree from the Bazel query

### CODE REVIEW

File: server/src/main/java/server/bazel/tree/WorkspaceTree.java [hunk 1]
```diff
@@ -16,6 +20,33 @@ public Node getRoot() {
...
     }
 
+    @Override
+    public String toString() {
+        return "WorkspaceTree{" +
+                "root=" + root +
+                '}';
+    }
+
+    public void clearBelowPath(String path) {
```
Comment (josiahsrc) [inline server/src/main/java/server/bazel/tree/WorkspaceTree.java:35]: docs

**Review by josiahsrc — APPROVED** [review]
Perf, one comment about function docs

File: server/src/main/java/server/bazel/tree/WorkspaceTree.java (hunk 1 above)
Comment (NovaMachina) [reply server/src/main/java/server/bazel/tree/WorkspaceTree.java:35]: Fixed

**Review by jasonxris — APPROVED** [review]
Looks good, i wish i had addSourceToTree and addTargetToTree to work with while writing my API tests.

**Pull request merged**